py3status/py3.py

```diff
@@ -1289,15 +1291,30 @@ def request(
         if timeout is None:
             timeout = getattr(self._py3status_module, "request_timeout", 10)
 
+        if retry_times is None:
+            retry_times = getattr(self._py3status_module, "request_retry_times", 3)
+
+        if retry_wait is None:
+            retry_wait = getattr(self._py3status_module, "request_retry_wait", 2)
+
         if "User-Agent" not in headers:
             headers["User-Agent"] = "py3status/{} {}".format(version, self._uid)
 
-        return HttpResponse(
-            url,
-            params=params,
-            data=data,
-            headers=headers,
-            timeout=timeout,
-            auth=auth,
-            cookiejar=cookiejar,
-        )
+        for n in range(1, retry_times + 1):
+            try:
+                return HttpResponse(
+                    url,
+                    params=params,
+                    data=data,
+                    headers=headers,
+                    timeout=timeout,
+                    auth=auth,
+                    cookiejar=cookiejar,
+                )
+            except (self.RequestTimeout, self.RequestURLError):
+                self.log("HTTP request retry {}/{}".format(n, retry_times))
+                if self.is_gevent():
+                    from gevent import sleep
+                else:
+                    from time import sleep
+                sleep(retry_wait)
```

Comment:
This part does not return failed HttpResponse like error code and stuff.

Reply:
I'm sorry I'm not sure I understand what you mean or want me to do with your comment?

Reply:
I'm sorry too. After looping few times, it will return `None` instead of `HttpResponse` which can cause exceptions in some modules. Try it with `coin_market`.

Reply:
got it sorry, fixed

Reply:
I think you're missing `sleep(retry_wait)` for the last `return get_http_response()` since it would be still failing.